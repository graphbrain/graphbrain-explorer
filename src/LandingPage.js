import React, { Component, Fragment } from 'react';

import { NavLink } from 'react-router-dom';

import {getData} from './api/getData';

import TopicsDropdown from './components/TopicsDropdown';

import Maps from './map/Maps';

import logo from './assets/graphbrain-logo.gif';

import './App.css';

class LandingPage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      chosenTopic: null,
      conflictsList: []
    }
  }

  componentDidMount() {
    getData('topics').then(data => {
      this.setState({
        conflictsList: data.viz_blocks[0].rows
      })

    })
  }

  handleSubmit() {
    if (this.state.chosenTopic) {
      window.location.href = `/map/?topic=${this.state.chosenTopic}`;
    }
  }

  handleInputChange(e) { 
    const { conflictsList } = this.state; 
    const topicLabel = e.currentTarget.value; 
    const chosenTopic = conflictsList.find(conflict => conflict.label === topicLabel); 
    const topicUrl = chosenTopic.url[0].slice(chosenTopic.url[0].lastIndexOf('/'));
    // console.log(conflictsList.find(conflict => conflict.label === topicLabel)); 
    this.setState({chosenTopic});
    // console.log(conflictsList.find(conflict => conflict.label === topicLabel))
    window.history.pushState({urlPath: topicUrl}, '', topicUrl); 

}

  handlePreDefinedClick() {
    this.setState({chosenTopic : "2"})
  }

  render () {
    const { chosenTopic } = this.state;
    return (
      <Fragment>
      {!chosenTopic && (
        <div className="App">
          <header className="App-header">
            <img src={logo} className="App-logo" alt="logo" />
            <p>
            Language, Knowledge, Cognition
            </p>
            <div className="buttonArea">
              <div className="dropdownArea">
                <TopicsDropdown 
                  handleSubmit={() => this.handleSubmit()} 
                  handleInputChange={(e) => this.handleInputChange(e)}
                  conflictsList={this.state.conflictsList}
                />
                {/* {this.state.chosenTopic && (
                <NavLink to={`/map?topic=${this.state.chosenTopic}`} className="navLivk"> Create Map </NavLink>) } */}
              </div>
              {/* <button className="navlink" onClick={() => this.handlePreDefinedClick()}> Pre-defined Map </button> */}
            </div>
          </header>
        </div>
      )}
      {chosenTopic && (
        <Maps chosenTopic={chosenTopic} handleBackClick={() => this.handleBackClick()}/>
      )}
      </Fragment>
    );
  }
}

export default LandingPage;
