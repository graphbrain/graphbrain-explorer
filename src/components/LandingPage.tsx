import React, { Fragment, useState, useEffect } from 'react';
import { NavLink } from 'react-router-dom';

import {getData} from '../api/getData';

import Maps from './map/Maps';
import Error from './Error'

import logo from '../assets/graphbrain-logo.gif';

import '../styles/landingPage.scss';

export interface Topic {
  id: string,
  label: string,
  url: string,
  weight: number
}

const LandingPage: React.FC<{}> = () => {

  const [chosenTopic, setChosenTopic] = useState<Topic | null>(null);
  const [topicsList, setTopicsList] = useState<Array<Topic>>([]);
  const [error, setError] = useState<boolean>(false)

  useEffect(() => {
    getData('topics').then(data => {
      if(data && data.viz_blocks) {
        const filteredTopics = data.viz_blocks[0].rows.filter((topic: Topic) => 
        topic.label.split(" ").length < 5 && topic.label.length !== 1
        ).sort((a:Topic, b:Topic) => (
          b.label < a.label ? 1 : -1
        ))
        setTopicsList(filteredTopics);
      }
      else {
        setError(true)
      }
    })
  }, [])

  const handleInputChange = (topic: Topic) => {
    setChosenTopic(topic);
  }

  // const handlePreDefinedClick = () => {
  //   setChosenTopic(2);
  // }

  return (
    <Fragment>
    {!chosenTopic && (
      <div className="landingWrapper">
        <aside className="aboutArea">
          <img src={logo} className="App-logo" alt="logo" />
          <div className="mainText">
            <p>Graphbrain is an Artificial Intelligence open-source software library and scientific research tool. It's aim is to facilitate automated meaning extraction and text understanding, as well as the exploration and inference of knowledge.</p>
            <p>This website is both a demonstration and an application of Graphbrain. It allows you to experience the news in a different way: instead of following the timelines of events as they unfold, it makes it possible to navigate the implicit networks of meaning that traverse temporal spans and publications. </p>
            <p> We invite you to explore how the various actors, topics, places and events are interconnected through relations of cooperation, common interest and conflict, through the various viewpoints present in the media.</p>
          </div>
        </aside>
          <div className="topicsArea">
            <h2 className="topicsTitle">Please choose a topic</h2>
            <nav className="topicsList">
            {topicsList.map(topic => (
            <NavLink 
              key={topic.id}
              onClick={() => handleInputChange(topic)}
              to={`${topic.url[0].slice(topic.url[0].lastIndexOf('/'))}`} 
              className="navLink"> {topic.label.length === 2 ? topic.label.toUpperCase() : topic.label} 
            </NavLink>
            ))}
            </nav>
          </div> 
            {/* <button className="navlink" onClick={() => this.handlePreDefinedClick()}> Pre-defined Map </button> */}
      </div>
    )}
    {chosenTopic && (
      <Maps />
    )}
    {error && <Error/>}
    </Fragment>
  );
}


export default LandingPage;
