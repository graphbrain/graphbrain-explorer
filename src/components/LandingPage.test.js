import React from "react";
import { render } from "@testing-library/react";

import '@testing-library/jest-dom/extend-expect';

import LandingPage from "./LandingPage";


describe("<LandingPage />", () => {
  it("renders correctly", () => {   
    const { asFragment } = render(<LandingPage/>);
    expect(asFragment()).toMatchSnapshot();   
  });

  it("renders the about area", async () => {
    const { getByTestId } = render(<LandingPage/>);
    expect(getByTestId('aboutArea')).toHaveTextContent('Artificial Intelligence');
  })
});