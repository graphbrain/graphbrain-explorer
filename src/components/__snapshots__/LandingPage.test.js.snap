// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="landingWrapper"
  >
    <aside
      class="aboutArea"
    >
      <img
        alt="logo"
        class="App-logo"
        src="graphbrain-logo.gif"
      />
      <div
        class="mainText"
      >
        <p>
          Graphbrain is an Artificial Intelligence open-source software library and scientific research tool. It's aim is to facilitate automated meaning extraction and text understanding, as well as the exploration and inference of knowledge.
        </p>
        <p>
          This website is both a demonstration and an application of Graphbrain. It allows you to experience the news in a different way: instead of following the timelines of events as they unfold, it makes it possible to navigate the implicit networks of meaning that traverse temporal spans and publications. 
        </p>
        <p>
           We invite you to explore how the various actors, topics, places and events are interconnected through relations of cooperation, common interest and conflict, through the various viewpoints present in the media.
        </p>
      </div>
    </aside>
    <div
      class="topicsArea"
    >
      <h2
        class="topicsTitle"
      >
        Please choose a topic
      </h2>
      <nav
        class="topicsList"
      />
    </div>
  </div>
</DocumentFragment>
`;
